woocommerce/woocommerce-gutenberg-products-block
# Fixes and improvements for the Price Slider Block

Resolves some related issues reported with the Price Slider. Tackled in one pull due to touching the same code for each.

1. Makes the size of the input based on the maxConstraint to avoid cutting off prices in the input boxes Fixes #2752
2. Prevents input values going below zero Fixes #2695 
3. Fixes keyboard input by removing the value from the range sliders Fixes #1679 
4. Adds aria value to read correct price without minor units Fixes #1657 
5. Fixes active filter display of prices so they match the slider
6. Resolves linting errors in price slider component

# How to test the changes in this Pull Request:

Add All Products, Price Filter, and Active Filters to a page.

1. Add a product to the store with a high price like 999999. View slider with inputs enabled. Price should not be cut off.
2. Confirm Active Filters display prices using the correct format (with commas for thousands etc)
3. Click a handle or tab to it and try using the keyboard to change the value of the slider. It will change. Focus will be lost after update.
4. With a screen reader, check what prices are announced when the slider has focus. It should be something like $100 rather than 10000 (without minor units).
5. Set min to 10 and max to 100, then set max to 0 via the inputs. Ensure min remains at 0 and does not go negative.



### Changelog

> Accessibility: Fixes keyboard input and screen reader text in the Price Slider Component.
> Fix: Prevent negative values in the price slider.
> Fix: Fixes formatting of prices in the active filters block and price slider inputs.
